maxim/tests/test_add_dataset_entries_comprehensive.py

```
@@ -0,0 +1,502 @@
+        self.dataset_id = "test-dataset-id"
+
+    def tearDown(self) -> None:
+        """Clean up after tests."""
+        if hasattr(Maxim, "_instance"):
+            delattr(Maxim, "_instance")
+        if "MAXIM_API_KEY" in os.environ:
+            del os.environ["MAXIM_API_KEY"]
+        if "MAXIM_BASE_URL" in os.environ:
+            del os.environ["MAXIM_BASE_URL"]
+
+    def _create_mock_response(self, content: bytes, headers: dict = None) -> Mock:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)